crates/core/src/contract/executor.rs

```diff
@@ -559,11 +581,54 @@ pub struct Executor<R = Runtime> {
     subscriber_summaries: HashMap<ContractKey, HashMap<ClientId, Option<StateSummary<'static>>>>,
     /// Attested contract instances for a given delegate.
     delegate_attested_ids: HashMap<DelegateKey, Vec<ContractInstanceId>>,
+    /// Tracks contracts that are being initialized and operations queued for them
+    contract_init_state: HashMap<ContractKey, ContractInitState>,
 
     event_loop_channel: Option<ExecutorToEventLoopChannel<ExecutorHalve>>,
 }
 
 impl<R> Executor<R> {
+    /// Process any queued operations for a contract that has completed initialization.
+    /// Returns the number of operations that were processed.
+    #[allow(dead_code)] // Will be used when we implement proper queue processing
+    pub(crate) async fn process_queued_operations(&mut self, key: ContractKey) -> usize
+    where
+        R: ContractExecutor,
+    {
+        if let Some(ContractInitState::Initializing { queued_ops, .. }) =
+            self.contract_init_state.remove(&key)
+        {
+            let count = queued_ops.len();
+            tracing::info!(
+                contract = %key,
+                operations = count,
+                "Processing queued operations for initialized contract"
+            );
+
+            for op in queued_ops {
+                let queue_duration = op.queued_at.elapsed();
+                tracing::debug!(
+                    contract = %key,
+                    queued_for_ms = queue_duration.as_millis(),
+                    "Processing queued operation"
+                );
+
+                // Process the operation through the executor
+                // Note: We can't call upsert_contract_state directly due to async recursion
```

Comment:
I don't think this is true, what is the issue, what is the compiler is complaining about? Just box the future and then it should work.

Reply:
This comment was in the process_queued_operations method which has been entirely removed, so this issue is resolved. The async recursion problem would have occurred if we tried to call upsert_contract_state from within itself, but since we removed the method that attempted this, the issue no longer exists.